src/components/LocationSelector/LocationSelectorDialog/LocationSelectorDialog.js

```diff
@@ -34,15 +36,26 @@ const LocationSelectorDialog = props => {
         onCloseModal,
     } = props;
 
+    function handleAddLocation(location) {
+        setTimeout(() => locationInputRef.current.focus());
+        onAddLocation(location);
+    }
+
+    function handleRemoveLocation(locationId) {
+        setTimeout(() => locationInputRef.current.focus());
```

Comment:
Can we skip timeout? Maybe add some hooks on `locationId` if it changes then focus input?

Reply:
@mukiienko is there something wrong with focusing input wrapped in setTimeout? We don't have an access to locationId here since it's an argument for event handling functions.

Reply:
Not transparent why the focus needs a timeout. It looks like a workaround for some problems with focusing. That code can cause problems in the future, for instance, in below code execution we need to add focus for another element, but it will not work because async focus rewrites sync focus, so we need for another focus add a timeout as well.